scm-webapp/src/main/java/sonia/scm/search/SearchableTypeResolver.java

```diff
@@ -62,26 +64,30 @@ public SearchableTypeResolver(PluginLoader pluginLoader) {
     fillMaps(convert(indexedTypes));
   }
 
-  private void fillMaps(Iterable<SearchableType> types) {
-    for (SearchableType type : types) {
+  private void fillMaps(Iterable<LuceneSearchableType> types) {
+    for (LuceneSearchableType type : types) {
       classToSearchableType.put(type.getType(), type);
       nameToClass.put(type.getName(), type.getType());
     }
   }
 
   @Nonnull
-  private Set<SearchableType> convert(Iterable<Class<?>> indexedTypes) {
+  private Set<LuceneSearchableType> convert(Iterable<Class<?>> indexedTypes) {
     return StreamSupport.stream(indexedTypes.spliterator(), false)
       .map(SearchableTypes::create)
       .collect(Collectors.toSet());
   }
 
-  public SearchableType resolve(Object object) {
+  public List<LuceneSearchableType> getSearchableTypes() {
+    return ImmutableList.copyOf(classToSearchableType.values());
```

Comment:
Do we really need a copy here or wouldn't a wrapping in `Collections.unmodifiableList` be sufficient?